server/service/src/report/report_service.rs

```diff
@@ -407,7 +406,14 @@ fn generate_report(
     if let Some(arguments) = arguments {
         context.insert("arguments", &arguments);
     }
+    println!("generating report!");
     let mut tera = tera::Tera::default();
+    let mut localisations = Localisations::new();
+    localisations.load_translations()?;
```

Comment:
You can pass through service provider to generate report method, it would be needed either way, also as commented previous load_translations can be in the ::new() method so that service provider initialisation remains in similar format